qemu/tests/x86_cpu_flag_intel_pt.py

```diff
@@ -0,0 +1,63 @@
+import time
+import logging
+
+from avocado.utils import process
+from virttest import error_context, env_process
+from provider.cpu_utils import check_cpu_flags
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu flag intel-pt.
+    1) Check if current flags are in the supported lists, if no, cancel test
+    2) Otherwise, set pt_mode = 1 on host at first
+    3) Boot guest with cpu model 'host' without intel-pt.
+    4) Check cpu flags in guest(only for linux guest)
+    5) For q35
+       5.1) boot guest with cpu model with intel-pt
+    6) For pc
+       6.1) boot guest with intel-pt and min-level=0x14
+    7) Check cpu flags in guest(only for linux guest)
+    8) Restore pt_mode value on host at last
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    flags = params["flags"]
+    check_cpu_flags(params, flags, test)
+
+    set_pt_mode = params.get("set_pt_mode")
+    get_pt_mode = params.get("get_pt_mode")
+    origin_value = process.getoutput(get_pt_mode).strip()
+
+    try:
+        if origin_value != '1':
+            process.system(set_pt_mode % '1', shell=True)
+        pt_mode = process.getoutput(get_pt_mode).strip()
+        if pt_mode != '1':
+            test.cancel("pt_mode can't be set to 1")
+
+        params["start_vm"] = "yes"
+        for x in range(2):
+            if x == 1:
+                params["auto_cpu_model"] = 'no'
+                params["cpu_model"] = 'host'
+                params["cpu_model_flags"] = ",enforce"
+
+            env_process.preprocess(test, params, env)
+            vm = env.get_vm(params["main_vm"])
+            error_context.context("Try to log into guest", logging.info)
+            session = vm.wait_for_login()
+            if params["os_type"] == "linux":
+                check_cpu_flags(params, flags, test, session)
+            vm.verify_kernel_crash()
+            session.close()
+            vm.destroy()
+    finally:
+        qemu_p = vm.get_pid()
+        while qemu_p:
```

Comment:
Why is this need? Shouldn't vm.destroy() have ended qemu process?

Reply:
@huangyum In case the machine is stuck which will cause the process of Line 57 to fail.

Reply:
@nanliu-r What about force to kill the process, in case it hang there here will lead to a end-less loop.

Reply:
vm.destroy(gracefully=False) should do all the work.